cmd/kubeadm/app/apis/kubeadm/v1beta1/types.go

```diff
@@ -126,6 +126,10 @@ type ControlPlaneComponent struct {
 
 	// ExtraVolumes is an extra set of host volumes, mounted to the control plane component.
 	ExtraVolumes []HostPathMount `json:"extraVolumes,omitempty"`
+
+	UID int64
+
+	GID int64
```

Comment:
i know this is a draft PR, but i'm not a fan of letting the user control this.
if we are going to migrate one component at at time to non-root that seems OK, but ideally we should just have all CP components migrate to rootless without API changes:
https://github.com/kubernetes/kubeadm/issues/1367